### DISCUSSION TITLE
For/While Loops in KCL

### DISCUSSION THREAD

**Discussion by jgomez720** [KittyCAD/modeling-app#647] [New Feature Requests]
### Description
This is for general loop use (everything outside of linear and circular patterns) in KCL

### Example code
This should output 5 separate boxes with different dimensions.
```
for (let i = 1; i <= 5; i++) {
  const box = startSketchAt([i,i])
    |> line([0, i], %)
    |> line([i, 0], %)
    |> line([0, -i], %)
    close(%)
    |> extrude(i)
}
```

**Comment by jessfraz** [comment]
Ill close mine hahaha https://github.com/KittyCAD/modeling-app/discussions/645

**Reply by jgomez720** [reply]
oh shit lol

**Reply by jessfraz** [reply]
all good

**Comment by jessfraz** [comment]
previous thoughts

- https://github.com/KittyCAD/kcl-experiments/issues/6#issuecomment-1641698307
- https://github.com/KittyCAD/modeling-app/issues/453

**Comment by adamchalmers** [comment]
IMO loops should be used for math algorithms. For duplicating geometry we should use patterns. Patterns are much faster because the engine can optimize and batch.
